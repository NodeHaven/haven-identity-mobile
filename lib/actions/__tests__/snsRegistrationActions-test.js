// Copyright (C) 2018 ConsenSys AG
//
// This file is part of uPort Mobile App.
//
// uPort Mobile App is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// uPort Mobile App is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with uPort Mobile App.  If not, see <http://www.gnu.org/licenses/>.
//
import {
  registerDeviceForNotifications,
  storeEndpointArn,
  receiveDeviceToken,
  storeDeviceToken,
  sendLocalNotification,
  updateBadgeCount,
  checkForNotifications,
  pollForNotifications,
  handleEncryptedMessage,
  initNotifications
 } from 'uPortMobile/lib/actions/snsRegistrationActions.js'

it('creates a REGISTER_DEVICE_FOR_NOTIFICATIONS action', () => {
  expect(registerDeviceForNotifications()).toMatchSnapshot()
})

it('creates a STORE_ENDPOINT_ARN action containing the arn', () => {
  expect(storeEndpointArn('aws:arn:dddd')).toMatchSnapshot()
})

it('creates a RECEIVE_DEVICE_TOKEN action containing the token', () => {
  expect(receiveDeviceToken('xxxxxx')).toMatchSnapshot()
})

it('creates a STORE_DEVICE_TOKEN action containing the token', () => {
  expect(storeDeviceToken('xxxxxx')).toMatchSnapshot()
})

it('creates a SEND_LOCAL_NOTIFICATION action containing the activity id and message', () => {
  expect(sendLocalNotification(123, 'You received a message')).toMatchSnapshot()
})

it('creates a UPDATE_BADGE_COUNT action', () => {
  expect(updateBadgeCount()).toMatchSnapshot()
})

it('creates a CHECK_FOR_NOTIFICATIONS action containing the token', () => {
  expect(checkForNotifications()).toMatchSnapshot()
})

it('creates a POLL_FOR_NOTIFICATIONS action containing the token', () => {
  expect(pollForNotifications()).toMatchSnapshot()
})

it('creates a HANDLE_ENCRYPTED_MESSAGE action containing the token', () => {
  expect(handleEncryptedMessage('012314')).toMatchSnapshot()
})

it('creates a INIT_NOTIFICATIONS action', () => {
  expect(initNotifications()).toMatchSnapshot()
})
