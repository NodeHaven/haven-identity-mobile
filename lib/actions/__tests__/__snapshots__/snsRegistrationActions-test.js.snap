// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creates a CHECK_FOR_NOTIFICATIONS action containing the token 1`] = `
Object {
  "type": "CHECK_FOR_NOTIFICATIONS",
}
`;

exports[`creates a HANDLE_ENCRYPTED_MESSAGE action containing the token 1`] = `
Object {
  "id": "012314",
  "type": "HANDLE_ENCRYPTED_MESSAGE",
}
`;

exports[`creates a INIT_NOTIFICATIONS action 1`] = `
Object {
  "type": "INIT_NOTIFICATIONS",
}
`;

exports[`creates a POLL_FOR_NOTIFICATIONS action containing the token 1`] = `
Object {
  "type": "POLL_FOR_NOTIFICATIONS",
}
`;

exports[`creates a RECEIVE_DEVICE_TOKEN action containing the token 1`] = `
Object {
  "deviceToken": "xxxxxx",
  "type": "RECEIVE_DEVICE_TOKEN",
}
`;

exports[`creates a REGISTER_DEVICE_FOR_NOTIFICATIONS action 1`] = `
Object {
  "type": "REGISTER_DEVICE_FOR_NOTIFICATIONS",
}
`;

exports[`creates a SEND_LOCAL_NOTIFICATION action containing the activity id and message 1`] = `
Object {
  "activityId": 123,
  "message": "You received a message",
  "repeatInterval": undefined,
  "type": "SEND_LOCAL_NOTIFICATION",
}
`;

exports[`creates a STORE_DEVICE_TOKEN action containing the token 1`] = `
Object {
  "deviceToken": "xxxxxx",
  "type": "STORE_DEVICE_TOKEN",
}
`;

exports[`creates a STORE_ENDPOINT_ARN action containing the arn 1`] = `
Object {
  "endpointArn": "aws:arn:dddd",
  "type": "STORE_ENDPOINT_ARN",
}
`;

exports[`creates a UPDATE_BADGE_COUNT action 1`] = `
Object {
  "type": "UPDATE_BADGE_COUNT",
}
`;
